### PR TITLE
Updates in matching 

### PR DESCRIPTION
**Pull request by VasilievaKA**
* Added default multi test method in AB test 
* Added group_matching 
* Added extended metrics in matching 
* Deleted two_sides parametr

### CODE REVIEW

File: hypex/reporters/matching.py [hunk 1]
```diff
@@ -1,23 +1,58 @@
...
+from hypex.utils import ExperimentDataEnum, ID_SPLIT_SYMBOL
 
 
 class MatchingDictReporter(DictReporter):
+    def __init__(self, searching_class: type = MatchingAnalyzer):
+        self.searching_class = searching_class
+        super().__init__()
+
+    @staticmethod
+    def _convert_dataset_to_dict(data: Dataset) -> Dict[str, Any]:
```
Comment (Dmatryus) [inline hypex/reporters/matching.py:16]: Without Adapter?

File: hypex/reporters/matching.py [hunk 2]
```diff
@@ -1,23 +1,58 @@
...
-        return self.extract_from_one_row_dataset(data.analysis_tables[analyzer_id])
+        return self._convert_dataset_to_dict(data.analysis_tables[analyzer_id])
+
+    @staticmethod
+    def _extract_from_additional_fields(data: ExperimentData):
+        indexes_id = data.get_one_id(
+            FaissNearestNeighbors, ExperimentDataEnum.additional_fields
+        )
+        return {
+            "indexes": "||".join(
```
Comment (Dmatryus) [inline hypex/reporters/matching.py:38]: Use splitter from utils

File: hypex/ui/matching.py [hunk 3]
```diff
@@ -29,5 +35,40 @@ def _extract_full_data(self, experiment_data: ExperimentData):
...
+                        experiment_data.ds[group_indexes_id] == group
+                    ].index,
+                    roles={"indexes": StatisticRole()},
+                )
+                for group, values in reformatted_resume.pop("indexes").items()
+            ]
+            indexes = indexes[0].append(indexes[1:]).sort()
+        else:
+            indexes = Dataset.from_dict(
+                {"indexes": list(map(int, resume["indexes"].split("||")))},
```
Comment (Dmatryus) [inline hypex/ui/matching.py:62]: import split from `utils`

**Review by Dmatryus — CHANGES_REQUESTED** [review]
Small changes are required

File: hypex/reporters/matching.py (hunk 1 above)
Comment (TonyKatkov89) [reply hypex/reporters/matching.py:16]: there are no ready suitable adapter

File: hypex/ui/matching.py (hunk 3 above)
Comment (TonyKatkov89) [reply hypex/ui/matching.py:62]: done

File: hypex/reporters/matching.py (hunk 2 above)
Comment (TonyKatkov89) [reply hypex/reporters/matching.py:38]: done

**Review by Dmatryus — APPROVED** [review]
🦾

**Pull request merged**